internal/pkg/aws/s3/s3.go

```diff
@@ -194,68 +195,96 @@ func FormatARN(partition, location string) string {
+	var prefixes []*s3.CommonPrefix
+	for _, output := range outputs {
+		contents = append(contents, output.Contents...)
+		prefixes = append(prefixes, output.CommonPrefixes...)
+	}
+	tree := treeprint.New()
+	// Add top-level files.
+	for _, object := range contents {
+		tree.AddNode(aws.StringValue(object.Key))
+	}
+	// Recursively add folders and their children.
+	if err := s.addNodes(tree, prefixes, bucket); err != nil {
+		return "", err
+	}
+	return tree.String(), nil
+}
+
+// BucketSizeAndCount returns the total size and number of objects in an S3 bucket.
+func (s *S3) BucketSizeAndCount(bucket string) (string, int, error) {
+	outputs, err := s.listObjects(bucket, "")
 	if err != nil {
-		if aerr, ok := err.(awserr.Error); ok && aerr.Code() == errCodeNotFound {
-			return false, nil
+		return "", 0, err
+	}
+	if outputs == nil {
+		return "", 0, nil
+	}
+	var size int64
+	var count int
+	for _, output := range outputs {
+		for _, object := range output.Contents {
+			size = size + aws.Int64Value(object.Size)
+			count = count + 1
 		}
-		return false, err
 	}
-
-	return true, nil
+	return humanize.Bytes(uint64(size)), count, nil
 }
 
-// GetBucketTree retrieves the objects in an S3 bucket and creates an ASCII tree representing their folder structure.
-func (s *S3) GetBucketTree(bucket string) (string, error) {
+func (s *S3) listObjects(bucket, delimiter string) ([]s3.ListObjectsV2Output, error) {
 	exists, err := s.bucketExists(bucket)
 	if err != nil {
-		return "", err
+		return nil, err
 	}
 	if !exists {
-		return "", nil
+		return nil, nil
 	}
-
-	var contents []*s3.Object
-	var prefixes []*s3.CommonPrefix
+	var outputs []s3.ListObjectsV2Output
 	listResp := &s3.ListObjectsV2Output{}
 	for {
 		listParams := &s3.ListObjectsV2Input{
 			Bucket:            aws.String(bucket),
-			Delimiter:         aws.String(slashDelimiter),
+			Delimiter:         aws.String(delimiter),
 			ContinuationToken: listResp.NextContinuationToken,
 		}
 		listResp, err = s.s3Client.ListObjectsV2(listParams)
 		if err != nil {
-			return "", fmt.Errorf("list objects for bucket %s: %w", bucket, err)
+			return nil, fmt.Errorf("list objects for bucket %s: %w", bucket, err)
 		}
-		contents = append(contents, listResp.Contents...)
-		prefixes = append(prefixes, listResp.CommonPrefixes...)
+		outputs = append(outputs, *listResp)
 		if listResp.NextContinuationToken == nil {
 			break
 		}
 	}
+	return outputs, nil
+}
 
-	tree := treeprint.New()
-	// Add top-level files.
-	for _, object := range contents {
-		tree.AddNode(aws.StringValue(object.Key))
+func (s *S3) bucketExists(bucket string) (bool, error) {
+	input := &s3.HeadBucketInput{
+		Bucket: aws.String(bucket),
 	}
-	// Recursively add folders and their children.
-	if err := s.addNodes(tree, prefixes, bucket); err != nil {
-		return "", err
+	_, err := s.s3Client.HeadBucket(input)
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok && aerr.Code() == errCodeNotFound {
+			return false, nil
+		}
```

Comment:
```suggestion
		var aerr *awserr.Error
		if errors.As(err, &aerr) && aerr.Code() == errCodeNotFound {
			return false, nil
		}
```
using `errors.As()` makes it so that any wrapped errors are caught as well!